src/main/scala/ldbc/snb/datagen/spark/generators/SparkPersonSerializer.scala

```diff
@@ -0,0 +1,126 @@
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.TaskContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
+
+import scala.collection.JavaConverters._
+
+import ldbc.snb.datagen.spark.util.FluentSyntax._
+
+object SparkPersonSerializer {
+
+  def apply(
+    persons: RDD[Person],
+    conf: LdbcConfiguration,
+    partitions: Option[Int] = None
+  )(implicit spark: SparkSession): Unit = {
+    val serializableHadoopConf = new SerializableConfiguration(spark.sparkContext.hadoopConfiguration)
+
+    persons
+      .withFoldLeft(partitions, (rdd: RDD[Person], p: Int) => rdd.repartition(p))
+      .foreachPartition(persons => {
+      val dynamicPersonSerializer = conf.getDynamicPersonSerializer
+      val hadoopConf = serializableHadoopConf.value
+      val partitionId = TaskContext.getPartitionId()
+      val buildDir = conf.getBuildDir
+
+      val fs = FileSystem.get(hadoopConf)
+      fs.mkdirs(new Path(buildDir))
+
+      dynamicPersonSerializer.initialize(
+        hadoopConf,
+        conf.getSocialNetworkDir,
+        partitionId,
+        conf.isCompressed,
+        conf.insertTrailingSeparator()
+      )
+
+      var insertEventSerializer: InsertEventSerializer = null
+      var deleteEventSerializer: DeleteEventSerializer = null
+
+      if ((DatagenParams.getDatagenMode eq DatagenMode.INTERACTIVE) || (DatagenParams.getDatagenMode eq DatagenMode.BI)) {
+        insertEventSerializer = new InsertEventSerializer(hadoopConf, conf.getBuildDir + "/temp_insertStream_person_" + partitionId, partitionId, DatagenParams.numUpdateStreams)
+        deleteEventSerializer = new DeleteEventSerializer(hadoopConf, conf.getBuildDir + "/temp_deleteStream_person_" + partitionId, partitionId, DatagenParams.numUpdateStreams)
+      }
+
+      try {
+        DatagenContext.initialize(conf)
+
+        for {p <- persons} {
+          if (DatagenParams.getDatagenMode eq DatagenMode.RAW_DATA) {
+            dynamicPersonSerializer.export(p)
+            for (k <- p.getKnows.iterator.asScala) {
+              dynamicPersonSerializer.export(p, k)
+            }
+          }
+          else {
+            if (p.getCreationDate < Dictionaries.dates.getBulkLoadThreshold && (p.getDeletionDate >= Dictionaries.dates.getBulkLoadThreshold && p.getDeletionDate <= Dictionaries.dates.getSimulationEnd)) {
+              dynamicPersonSerializer.export(p)
+              if (p.isExplicitlyDeleted) {
+                deleteEventSerializer.export(p)
+                deleteEventSerializer.changePartition()
+              }
+            }
+            else if (p.getCreationDate < Dictionaries.dates.getBulkLoadThreshold && p.getDeletionDate > Dictionaries.dates.getSimulationEnd) dynamicPersonSerializer.export(p)
+            else if (p.getCreationDate >= Dictionaries.dates.getBulkLoadThreshold && (p.getDeletionDate >= Dictionaries.dates.getBulkLoadThreshold) && p.getDeletionDate <= Dictionaries.dates.getSimulationEnd) {
```

Comment:
nitpicking: the paren should close at the end. (doesn't influence the result though)